common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -27,8 +29,40 @@ class EntitlementViewSet(viewsets.ModelViewSet):
     def get_queryset(self):
         user = self.request.user
         if user.is_staff:
-            return CourseEntitlement.objects.all().select_related('user')
-        return CourseEntitlement.objects.filter(user=user).select_related('user')
+            return CourseEntitlement.objects.all().select_related('user').select_related('enrollment_course_run')
+        return CourseEntitlement.objects.filter(user=user).select_related('user').select_related(
+            'enrollment_course_run'
+        )
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        Override the retrieve method to expire a record that is past the
+        policy and is requested via the API before returning that record.
+        """
+        entitlement = self.get_object()
+        entitlement.update_expired_at()
+        serializer = self.get_serializer(entitlement)
+        return Response(serializer.data)
+
+    def list(self, request, *args, **kwargs):
+        """
+        Override the list method to expire records that are past the
+        policy and requested via the API before returning those records.
+        """
+        queryset = self.filter_queryset(self.get_queryset())
+        user = self.request.user
+        if not user.is_staff:
```

Comment:
This logic will prevent staff users from being able to see their own entitlements when they visit pages that make use of this endpoint. Instead, it will return all the entitlements in the database.

Reply:
I think we should probably treat staff users the same as everyone else. If we need an admin endpoint where staff users can see the list of all entitlements, I think that should be separate from this.

@staubina what do you think?

Reply:
I agree.  We can also just add it to the user filter and let staff see all of them.

Reply:
I'm on board, though the original logic wasn't implemented as part of this PR. Personally I'd rather see this fixed and tracked in a story separate from LEARNER-2983, though not sure how you guys feel?